src/plugins/data/common/search/search_source/fetch/request_error.ts

```diff
@@ -16,9 +16,9 @@ import { SearchError } from './types';
  * @param {Object} resp - optional HTTP response
  */
 export class RequestFailure extends KbnError {
-  public resp?: estypes.SearchResponse<any>;
-  constructor(err: SearchError | null = null, resp?: estypes.SearchResponse<any>) {
-    super(`Request to Elasticsearch failed: ${JSON.stringify(resp || err?.message)}`);
+  public resp?: IKibanaSearchResponse;
+  constructor(err: SearchError | null = null, resp?: IKibanaSearchResponse) {
+    super(`Request to Elasticsearch failed: ${JSON.stringify(resp?.rawResponse || err?.message)}`);
 
     this.resp = resp;
```

Comment:
We can avoid repeating the declaration in the future.

```suggestion
  constructor(err: SearchError | null = null, public resp?: IKibanaSearchResponse) {
    super(`Request to Elasticsearch failed: ${JSON.stringify(resp?.rawResponse || err?.message)}`);
```